### PR TITLE
Document parameterized jobs and dispatch command

### PR DESCRIPTION
**Pull request by dadgar**
This PR:

* documents parameterized jobs, the dispatch command and the appropriate APIs.
* Renames `dispatch_input` to `dispatch_payload`
* Removes the HCL only meta_keys wrapper inside the `parameterized` block and renames the required and optional keys.

### CODE REVIEW

File: command/job_dispatch.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ General Options:
 
 Dispatch Options:
 
+  -meta <key>=<value>
+    Meta takes a key/value pair seperated by "=". The metadata key will be
+    injected into the job's metadata. The flag can be provided more than
+    once to inject multiple metadata key/value pairs. Arbitrary injection is not
+    allowed. The parameterized job must allow the key to be injected.
```
Comment (armon) [inline command/job_dispatch.go]: Can we document what the behavior is for a default value? Does this overwrite the existing value?

Comment (armon) [inline command/job_dispatch.go]: I don't like the work `injected` can we use `merged` instead?

File: website/source/docs/commands/job-dispatch.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  The dispatch command is used to create an instance of a parameterized job.
+---
+
+# Command: job dispatch
+
+~> The `job dispatch` subcommand described here is available only in version
+0.5.3 and later. The release canidate is downloadable on the [releases
+page.](https://releases.hashicorp.com/nomad/0.5.3-rc1/)
+
+The `job dispatch` command is used to create new instances of a [parameterized
```
Comment (armon) [inline website/source/docs/commands/job-dispatch.html.md.erb]: I feel like the command and the `parameterized job` should link somewhere?

File: website/source/docs/commands/job-dispatch.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+job]. The parameterized job captures a job's configuration and runtime
+requirements in a generic way and `dispatch` is used to provide the input for
+the job to run against. One can think of the parameterized job as a function
```
Comment (armon) [inline website/source/docs/commands/job-dispatch.html.md.erb]: Replace "One can think of the parameterized job as a function definition and dispatch is used to invoke the function." with "A parameterized job is similar to a function definition, and dispatch is used to invoke the function." Simpler language.

File: website/source/docs/commands/job-dispatch.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+caller to track the status of the job, much like a future or promise in some
+programming languages.
+
+## Usage
+
+```
+nomad job dispatch [options] <parameterized job> [input source]
+```
+
+Dispatch creates an instance of a parameterized job. A data payload to the
```
Comment (armon) [inline website/source/docs/commands/job-dispatch.html.md.erb]: What is the size limitation of the input? I would document that here

File: website/source/docs/http/job.html.md [hunk 5]
```diff
@@ -336,6 +340,52 @@ region is used; another region can be specified using the `?region=` query param
...
+  <dt>URL</dt>
+  <dd>`/v1/job/<ID>/dispatch`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    <ul>
+      <li>
+        <span class="param">Payload</span>
+        <span class="param-flags">optional</span>
+        A `[]byte` array encoded as a base64 string.
```
Comment (armon) [inline website/source/docs/http/job.html.md]: Size limitation here too

File: website/source/docs/http/json-jobs.html.md [hunk 6]
```diff
@@ -194,6 +207,24 @@ The `Job` object supports the following keys:
...
+  be dispatched against. The `ParamaterizedJob` object supports the following
+  attributes:
+
+  * `MetaOptional` - Specifies the set of metadata keys that may be provided
+    when dispatching against the job as a string array.
+
+  * `MetaRequired` - Specifies the set of metadata keys that must be provided
+    when dispatching against the job as a string array.
+
+  * `Payload` - Specifies the requirement of providing a payload when
```
Comment (armon) [inline website/source/docs/http/json-jobs.html.md]: What is the default value?

File: website/source/docs/job-specification/parameterized.html.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+## `parameterized` Parameters
+
+- `meta_optional` `(array<string>: nil)` - Specifies the set of metadata keys that
+   may be provided when dispatching against the job.
+
+- `meta_required` `(array<string>: nil)` - Specifies the set of metadata keys that
+  must be provided when dispatching against the job.
+
+- `payload` `(string: "optional")` - Specifies the requirement of providing a
```
Comment (armon) [inline website/source/docs/job-specification/parameterized.html.md]: What is the default? Maximum size?

File: website/source/docs/commands/job-dispatch.html.md.erb (hunk 2 above)
Comment (dadgar) [inline website/source/docs/commands/job-dispatch.html.md.erb]: You are on the command page and it is kinda weird but that is the syntax for linking. [parameterized job] there links to the correct job-spec page.

File: website/source/docs/job-specification/parameterized.html.md (hunk 7 above)
Comment (dadgar) [inline website/source/docs/job-specification/parameterized.html.md]: The default is in the parenthesis.

**Pull request merged**